debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCArrayletObjectModel_V2.java

```diff
@@ -67,4 +69,43 @@ public UDATA getTotalFootprintInBytesWithHeader(J9IndexableObjectPointer arrayPt
 		UDATA totalFootprint = spineSize.add(externalArrayletSize);
 		return totalFootprint;
 	}
+
+	/**
+	 * Determine the validity of the data address belonging to arrayPtr.
+	 *
+	 * @param arrayPtr array object who's data address validity we are checking
+	 * @throws CorruptDataException if there's a problem accessing the indexable object dataAddr field
+	 * @return true if the data address of arrayPtr is valid, false otherwise
+	 */
+	@Override
+	public boolean hasCorrectDataAddrPointer(J9IndexableObjectPointer arrayPtr) throws CorruptDataException
+	{
+		boolean hasCorrectDataAddrPointer;
+		try {
+			UDATA dataSizeInBytes = getDataSizeInBytes(arrayPtr);
+			VoidPointer dataAddr = J9IndexableObjectHelper.getDataAddrForIndexable(arrayPtr);
+			boolean isValidDataAddrForOffHeapObject = isIndexableObjectOffHeap(dataAddr, dataSizeInBytes);
+
+			if (dataSizeInBytes.isZero()) {
+				VoidPointer discontiguousDataAddr = VoidPointer.cast(arrayPtr.addOffset(J9IndexableObjectHelper.discontiguousHeaderSize()));
+				hasCorrectDataAddrPointer = (dataAddr.isNull() || dataAddr.equals(discontiguousDataAddr));
```

Comment:
We could be more strict if enableVirtualLargeObjectHeap is true, so that we disallow null.
Null is only possible in interim builds that don't have offheap code or in new builds with offheap disabled.

Reply:
actually, my comment was not correct. at least for interpreter, we will leave it null.